File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.cpp
int init=[] {ios_base::sync_with_stdio(false);cin.tie(0);return 0;}();
class Solution {
public:
    string removeStars(string s) {
        vector<int> stack;
        string str="";
        
        for (char chr : s) {
            if (chr == '*') stack.pop_back();
            else stack.push_back(chr);
        }
        
        for (int i=0;i<stack.size();++i) {
            str+=stack[i];
        }
        return str;
    }
};

File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.cs
public class Solution {
    public string RemoveStars(string s) {
        List<char> stack = new List<char>();

        foreach (char c in s) {
            if (c == '*') {
                if (stack.Count > 0) {
                    stack.RemoveAt(stack.Count - 1);
                }
            } else {
                stack.Add(c);
            }
        }

        return new string(stack.ToArray());
    }
}